### PR TITLE
[xcbuild] Split Libraries/Specifications into separate install components

### PR DESCRIPTION
**Pull request by Ktwu**
Using components allows builds to install only the stuff it needs.

### CODE REVIEW

File: Libraries/acdriver/CMakeLists.txt [hunk 1]
```diff
@@ -19,11 +19,11 @@ add_library(acdriver SHARED
 
 target_link_libraries(acdriver PUBLIC xcassets util plist ext)
 target_include_directories(acdriver PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/Headers")
-install(TARGETS acdriver DESTINATION usr/lib)
```
Comment (strager) [inline Libraries/acdriver/CMakeLists.txt]: Instead, push/set/pop `CMAKE_INSTALL_DEFAULT_COMPONENT_NAME` in the `Libraries/CMakeLists.txt` (or whatever calls `add_subdirectory`). This reduces the diff and is less nasty (though is action-at-a-distance).

Comment (Ktwu) [inline Libraries/acdriver/CMakeLists.txt]: Sure, that seems reasonable.

**Comment by Ktwu** [comment]
Any feedback?  Is this a good idea?  Bad idea?

**Comment by strager** [comment]
From a CMake language perspective, I think this diff looks good. I'll let a maintainer approve.

Why are the specifications optionally installed?

**Comment by Ktwu** [comment]
> Why are the specifications optionally installed?

What do you mean optionally installed?  If no component is specified, all are installed by default.

**Comment by strager** [comment]
A CMake user can choose to not install specifications. Why would a CMake user do that?

**Comment by Ktwu** [comment]
I know little about these specifications, but they appear related to Xcode, right?  I don't have Xcode on Linux, my target environment, so I don't care about specifications.

**Comment by grp** [comment]
Specifications are used by xcbuild (and Xcode) in order to resolve build settings into command line flags. If you want to do builds on Linux using xcbuild, you'll need a set of specifications — either the ones from Xcode or the (still incomplete) set from xcbuild.

**Comment by Ktwu** [comment]
Ok, that's good to know.  For the moment, all I actually care about are some of the tools inside of xcbuild (plutil on Linux, for example) -- I don't need specifications for each of the individual tools, do I?

**Comment by grp** [comment]
Nope, just for building.

**Comment by Ktwu** [comment]
Cool.  So, to answer strager's question, I care about this because I actually only want xcbuild's tooling.  I'm not using xcbuild itself because I'm trying to integrate the use of these tools with build structures set up with BUCK.

**Comment by strager** [comment]
Mmk, lgtm.

File: Libraries/acdriver/CMakeLists.txt (hunk 1 above)
Comment (grp) [inline Libraries/acdriver/CMakeLists.txt]: I kinda like the way @Ktwu did it first, but not worth extra changes now.

**Pull request closed without merging**